src/accountsdb/account_store.zig

```diff
@@ -18,6 +18,32 @@ const AccountSharedData = sig.runtime.AccountSharedData;
 
 const AccountsDB = accounts_db.AccountsDB;
 
+/// Interface for both reading and writing accounts.
+///
+/// Do not use this unless you need to *write* into the database generically.
+/// Otherwise use AccountReader if you only need to read accounts.
+pub const AccountStore = union(enum) {
```

Comment:
I think it would make sense to make these unions all share a tag, both to illustrate the fact that they share a similar purpose in acting as "views" into an accountsdb-looking-shape, and also for the bonus that it keeps them in sync, both in terms of the names and the order of declaration.
```zig
pub const ViewKind = enum { accounts_db, thread_safe_map, noop };
pub const AccountStore = union(ViewKind) {...};
pub const AccountReader = union(ViewKind) {...};
pub const SlotAccountReader = union(ViewKind) {...};
```